crates/nu-protocol/src/config/table.rs

```diff
@@ -424,6 +433,19 @@ impl UpdateFromValue for TableConfig {
                     Ok(val) => self.missing_value_symbol = val.to_string(),
                     Err(_) => errors.type_mismatch(path, Type::String, val),
                 },
+                "batch_duration" => match val.as_duration() {
+                    Err(_) => errors.type_mismatch(path, Type::Duration, val),
+                    Ok(..0) => errors.invalid_value(path, "a non-negative duration", val),
+                    Ok(val) => self.batch_duration = Duration::from_nanos(val as u64),
+                },
```

Comment:
```suggestion
                "batch_duration" => {
                    match Duration::from_value(val.clone()).map_err(ConfigError::from) {
                        Ok(val) => self.batch_duration = val,
                        Err(err) => errors.error(err),
                    }
                }
```
`Duration` implements `FromValue` so this would work too. The error message for negative values are a bit different though:
```
Error: nu::shell::invalid_config

  × Encountered 1 error(s) when updating config

Error: nu::shell::needs_positive_value

  × Negative value passed when positive one is required
   ╭─[entry #2:1:36]
 1 │ $env.config.table.batch_duration = -1sec
   ·                                    ──┬──
   ·                                      ╰── use a positive value
   ╰────
```

Of course we can adjust the error message for `Duration::from_value` to make it better. IIRC it's currently only used in `watch`.

Reply:
That looks great 👍🏻